pkg/polaris/polaris_test.go

```diff
@@ -990,3 +994,590 @@ func TestGcpPermissions(t *testing.T) {
+// namespace snapshot
+func TestTakeK8NamespaceSnapshot(t *testing.T) {
+	testClusterID := uuid.MustParse("2a47d1f1-0236-4030-87d4-837d2a75b370")
+	ctx := context.Background()
+	testServiceAccount := &ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
+		Name:           "kupatest",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+	testServiceAccount, err := DefaultServiceAccount(true)
+
+	client, err := NewClient(ctx, testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(nss) == 0 {
+		t.Fatal("No namespace found")
+	}
+
+	ns := nss[0] // get first namespace
+	info, err := client.K8s().TakeK8NamespaceSnapshot(ctx, ns.ID, "00000000-0000-0000-0000-000000000000")
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("%v\n", info)
+}
+
+// TestGetTaskchainInfo verifies that the SDK can fetch taskchain states
+func TestGetTaskchainInfo(t *testing.T) {
+	testClusterID := uuid.MustParse("2a47d1f1-0236-4030-87d4-837d2a75b370")
+	ctx := context.Background()
+	testServiceAccount := &ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
+		Name:           "kupatest",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+	testServiceAccount, err := DefaultServiceAccount(true)
+
+	client, err := NewClient(ctx, testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(nss) == 0 {
+		t.Fatal("No namespace found")
+	}
+
+	ns := nss[1] // get first namespace
+	info, err := client.K8s().TakeK8NamespaceSnapshot(ctx, ns.ID, "00000000-0000-0000-0000-000000000000")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 6; i++ {
+		tk, err := client.K8s().GetTaskchainInfo(ctx, info.TaskchainId, "snapshot-namespace")
+		if err == nil {
+			fmt.Printf("%v\n", tk)
+			return
+		}
+		if !strings.Contains(err.Error(), "Taskchain does not exist") ||
+			!strings.Contains(err.Error(), "State key not found in response") ||
+			!strings.Contains(err.Error(), "Taskchain still in wait state") {
+			fmt.Printf("%v : %v\n", info, err)
+			time.Sleep(15 * time.Second)
+		} else {
+			t.Fatal(err.Error())
+		}
+	}
+
+	t.Fatal("Task chain does not start")
+}
+
+// TestGetActivitySeriesConnection verifies that the SDK can fetch events
+func TestGetActivitySeriesConnection(t *testing.T) {
+	ctx := context.Background()
+	testServiceAccount := &ServiceAccount{
+		ClientID:       "client|gYtGLLyhcp3rI02kkbIdZt7CMFzIhg54",
+		ClientSecret:   "JQfCRwuiiwzB_7Ibt1UIOcAT0wQTYnRNF2ikPD8aTaTOqbiYhfm8v7Lb6pY6NBis",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: High <!-- a695ae80-4342-430b-bf40-53b187eb0a9e -->